docs/static/transforming-data.asciidoc

```diff
@@ -0,0 +1,520 @@
+
+Parses key-value pairs.
++
+For example, let's say you have a log message that contains the following
+key-value pairs: `ip=1.2.3.4 error=REFUSED`.
++
+The following config parses the key-value pairs into fields:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  kv { }
+}
+--------------------------------------------------------------------------------
++
+After the filter is applied, the event in the example will have these fields:
++
+* `ip: 1.2.3.4`
+* `error: REFUSED`
+
+
+<<plugins-filters-grok,grok filter>>::
+
+Parses unstructured event data into fields. This tool is perfect for syslog
+logs, Apache and other webserver logs, MySQL logs, and in general, any log
+format that is generally written for humans and not computer consumption.
+Grok works by combining text patterns into something that matches your
+logs.
++
+For example, let's say you have an HTTP request log that contains
+the following message:
++
+[source,json]
+--------------------------------------------------------------------------------
+55.3.244.1 GET /index.html 15824 0.043
+--------------------------------------------------------------------------------
++
+The following config parses the message into fields:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  grok {
+    match => { "message" => "%{IP:client} %{WORD:method} %{URIPATHPARAM:request} %{NUMBER:bytes} %{NUMBER:duration}" }
+  }
+}
+--------------------------------------------------------------------------------
++
+After the filter is applied, the event in the example will have these fields:
++
+* `client: 55.3.244.1`
+* `method: GET`
+* `request: /index.html`
+* `bytes: 15824`
+* `duration: 0.043`
+
+[[lookup-enrichment]]
+=== Enriching Data with Lookups
+
+The plugins described in this section are useful for enriching data with
+additional info, such as GeoIP and user agent info.
+
+Also see <<filter-plugins>> for the full list of available filter plugins. 
+
+<<plugins-filters-dns,dns filter>>::
+
+Performs a standard or reverse DNS lookup.
++
+The following config performs a reverse lookup on the address in the
+`source_host` field and replaces it with the domain name:
++
+[source,json]
+--------------------------------------------------------------------------------
+filter {
+  dns {
+    reverse => [ "source_host" ]
+    action => "replace"
+  }
+}
+--------------------------------------------------------------------------------
+
+    
+<<plugins-filters-elasticsearch,elasticsearch>>::
+
+Copies fields from previous log events in Elasticsearch to current events. 
++
+[source,json]
+--------------------------------------------------
+NEED A SIMPLE EXAMPLE HERE. 
+--------------------------------------------------
+
+
+//REVEIWERS: I NEED A BETTER EXAMPLE FOR THE ELASTICSEARCH FILTER. THE EXAMPLE IN THE DOCUMENTATION MIGHT BE TOO COMPLEX FOR AN OVERVIEW.
```

Comment:
Pinging @gingerwizard who has experience using this filter. Perhaps he has some ideas for a simple example here.